src/realm/list.hpp

```diff
@@ -854,7 +854,8 @@ inline bool LnkLst::is_null(size_t ndx) const
 inline Mixed LnkLst::get_any(size_t ndx) const
 {
     update_if_needed();
-    return m_list.get_any(virtual2real(ndx));
+    auto obj_key = m_list.get(virtual2real(ndx));
+    return ObjLink{get_target_table()->get_key(), obj_key};
```

Comment:
Does this not defeat the purpose of the `Mixed` return type?

Reply:
You create a Mixed from an ObjLink. It is done implicitly. It has the advantage that you can get both an ObjKey and an ObjLink out of the Mixed.

Reply:
This is just fixing an inconsistency that caused problems with these changes: Dictionary's get_any() gave a typed link and LnkLst's gave an untyped link, which made sense given the history of the functions but complicated things for code trying to call get_any() on an arbitrary Collection.